config/config_test.go

```diff
@@ -41,3 +32,65 @@ func TestProctorURL(t *testing.T) {
 		assert.NoError(t, err)
 	}
 }
+
+func TestProctorEmailId(t *testing.T) {
+	proctorConfigFileExistedBeforeTest, proctorConfigFilePath, existingConfigFileData, proctorConfigDir := readProctorConfigFile()
+
+	EmailId := []byte("EMAIL_ID: foobar@gmail.com")
+	err := ioutil.WriteFile(proctorConfigFilePath, EmailId, 0644)
+	assert.NoError(t, err)
+
+	config.InitConfig()
+	configuredEmailId := config.EmailId()
+
+	assert.Equal(t, "foobar@gmail.com", configuredEmailId)
+
+	if proctorConfigFileExistedBeforeTest {
+		err = ioutil.WriteFile(proctorConfigFilePath, existingConfigFileData, 0644)
+		assert.NoError(t, err)
+	} else {
+		err = os.Remove(proctorConfigFilePath)
+		assert.NoError(t, err)
+
+		err = os.Remove(proctorConfigDir)
+		assert.NoError(t, err)
+	}
+}
+
+func TestProctorAccessToken(t *testing.T) {
+	proctorConfigFileExistedBeforeTest, proctorConfigFilePath, existingConfigFileData, proctorConfigDir := readProctorConfigFile()
+
+	AccessToken := []byte("ACCESS_TOKEN: access-token")
+	err := ioutil.WriteFile(proctorConfigFilePath, AccessToken, 0644)
+	assert.NoError(t, err)
+
+	config.InitConfig()
+	configuredAccessToken := config.AccessToken()
+
+	assert.Equal(t, "access-token", configuredAccessToken)
+
+	if proctorConfigFileExistedBeforeTest {
+		err = ioutil.WriteFile(proctorConfigFilePath, existingConfigFileData, 0644)
+		assert.NoError(t, err)
+	} else {
+		err = os.Remove(proctorConfigFilePath)
+		assert.NoError(t, err)
+
+		err := os.Remove(proctorConfigDir)
+		assert.NoError(t, err)
+	}
+}
+
+func readProctorConfigFile() (proctorConfigFileExistedBeforeTest bool, proctorConfigFilePath string, existingConfigFileData []byte, proctorConfigDir string) {
```

Comment:
NOTE: Optional suggestion.
In `config.go`, the path to config file is hard-coded as `home := "$HOME/.proctor"` 

If you want, you can make it configurable for test env as 
```
if ENV['test'] == true
  home := /tmp/proctor.yaml
else
  home := "$HOME/.proctor"
end
```

This will help you get rid of this complex tear down process in cases proctor config file was already present.

Reply:
Done, refactored the file creation and deletion logic in specs.